src/main/java/frc/robot/subsystems/reef/rollers/Rollers.java

```diff
@@ -0,0 +1,91 @@
+package frc.robot.subsystems.reef.rollers;
+
+import edu.wpi.first.math.filter.LinearFilter;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Rollers extends SubsystemBase {
+  private final RollersIO io;
+  private final RollersIOInputsAutoLogged inputs;
+  private final LinearFilter velocityFilter = LinearFilter.movingAverage(5);
+  private final LinearFilter currentFilter = LinearFilter.movingAverage(5);
+
+  public Rollers(RollersIO io, RollersIOInputsAutoLogged inputs) {
+    this.io = io;
+    this.inputs = inputs;
+  }
+
+  @Override
+  public void periodic() {
+    io.updateInputs(inputs);
+  }
+
+  private void runRollers(double volts) {
+    io.runVolts(volts);
+  }
+
+  private void stopRollers() {
+    io.stop();
+  }
+
+  public boolean coralHeld() {
+    double filteredVelocity = velocityFilter.calculate(Math.abs(inputs.velocityRotsPerSec));
+    double filteredStatorCurrent = currentFilter.calculate(inputs.currentAmps);
+    return (filteredVelocity <= RollersConstants.coralVelocityThreshold
+            && (filteredStatorCurrent >= RollersConstants.coralCurrentThreshold)
+        || filteredStatorCurrent <= -2);
+  }
+
+  public boolean algaeHeld() {
+    double filteredVelocity = velocityFilter.calculate(Math.abs(inputs.velocityRotsPerSec));
+    double filteredStatorCurrent = currentFilter.calculate(inputs.currentAmps);
+    return (filteredVelocity <= RollersConstants.algaeVelocityThreshold
+            && (filteredStatorCurrent >= RollersConstants.algaeCurrentThreshold)
+        || filteredStatorCurrent <= -2);
+  }
+
+  public Command placeCoralAndHighAlgae() {
```

Comment:
Consistent names matter!
```suggestion
  public Command placeCoralAndHighDealgify() {
```